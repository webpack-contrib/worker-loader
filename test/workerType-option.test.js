import path from 'path';

import del from 'del';

import {
  compile,
  getCompiler,
  getErrors,
  getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from './helpers';

describe('"workerType" option', () => {
  beforeAll(() => del(path.resolve(__dirname, `outputs`)));

  it('should use "Worker"', async () => {
    const compiler = getCompiler('./basic/entry.js');
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work the "SharedWorker" value', async () => {
    const compiler = getCompiler('./basic/entry.js', {
      workerType: 'SharedWorker',
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work the "ServiceWorker" value', async () => {
    const compiler = getCompiler('./basic/entry.js', {
      workerType: 'ServiceWorker',
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
