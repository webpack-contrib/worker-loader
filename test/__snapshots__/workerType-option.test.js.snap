// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"workerType" option should use "Worker": errors 1`] = `Array []`;

exports[`"workerType" option should use "Worker": module 1`] = `
"module.exports = function() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
};"
`;

exports[`"workerType" option should use "Worker": result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should use "Worker": warnings 1`] = `Array []`;

exports[`"workerType" option should work the "ServiceWorker" value: errors 1`] = `Array []`;

exports[`"workerType" option should work the "ServiceWorker" value: module 1`] = `
"module.exports = function() {
  return new ServiceWorker(__webpack_public_path__ + \\"test.worker.js\\");
};"
`;

exports[`"workerType" option should work the "ServiceWorker" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should work the "ServiceWorker" value: warnings 1`] = `Array []`;

exports[`"workerType" option should work the "SharedWorker" value: errors 1`] = `Array []`;

exports[`"workerType" option should work the "SharedWorker" value: module 1`] = `
"module.exports = function() {
  return new SharedWorker(__webpack_public_path__ + \\"test.worker.js\\");
};"
`;

exports[`"workerType" option should work the "SharedWorker" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"workerType" option should work the "SharedWorker" value: warnings 1`] = `Array []`;
